### DISCUSSION TITLE
Reading of two Neey Balancer with ESP32 By Savio

### DISCUSSION THREAD

**Discussion by saviothecnic** [syssi/esphome-jk-bms#392] [General]
Reading of two Neey Balancer with ESP32 By Savio

I wasted a lot of time on it and with the help of my friend Vincenzo
I managed to read two Neey balancers with a single ESP32
If you have anyone you need or if you want to add them among the examples
that work this is the yaml I used
Rename by removing .txt since otherwise it wouldn't attach

[Esp32-Heltec-Balancer.yaml.txt](https://github.com/syssi/esphome-jk-bms/files/13323737/Esp32-Heltec-Balancer.yaml.txt)

**Comment by TEETH33** [comment]
IS THE NEEY the 4th generation one? I think so
I would like to add it to HOMASS, esp32 must have what characteristics? hello and thanks great work

**Reply by saviothecnic** [reply]
Yes I am currently using it with 5 neey
maximum two for Esp32 is working fine
both with 4 generation Neey
but also with a 3 generation neey that I got many years ago video https://youtu.be/gc0NTUYBpK4?si=GMFIaUeeB8XUclvI

**Comment by syssi** [comment]
Any ESP32 development board should work. May be very recent ESP32 models should be avoided as long they aren't supported by ESPHome.

**Comment by TEETH33** [comment]
salve e posdibile avere ik link del file yaml per esphome grazie

**Reply by saviothecnic** [reply]
A inizio discussione ho messo il codice che ho usato io e vedi nel video

**Comment by DougKremer** [comment]
I expanded the multiple device example to three and it is working perfectly on a single ESP32 via bluetooth to read three NEEY 4th gen balancers.

**Reply by TommyZog** [reply]
Hello, could i become your yaml for three Neey?

**Reply by DougKremer** [reply]
Yep this is my YAML:

[Three NEEY balancers.yaml](https://github.com/user-attachments/files/22259889/Three.NEEY.balancers.yaml)

**Reply by TommyZog** [reply]
Thank you for your help.
My yaml for 1 Neey and for 2 Neey are working with no problems, but my yaml for 3 Neeys make some problems ( i think i missed something, but now i can compare)

**Reply by DougKremer** [reply]
You are welcome.  I should have added it to begin with.  I probably didn't know how, I was very new.

**Reply by TommyZog** [reply]
I think 3 Neey is too much for the ESP. mqtt is no problem, but in HA there is no data, if i take 3 devices.

**Reply by DougKremer** [reply]
Hmm, it’s working fine on mine.  Make sure you have the Bluetooth monitoring switches enabled. Maybe my esps have more horsepower?

**Reply by TommyZog** [reply]
switched from ESP-WROOM-32 to ESP32-C6 and now it works.
Nothing can replace horsepower ;-)
Thanks for help

**Comment by TommyZog** [comment]
Does anyone here have a nice integration with Home Assistant?
